extensions/nginx/index.js

```diff
@@ -132,23 +132,24 @@ class NginxExtension extends cli.Extension {
         }, {
             title: 'Preparing nginx for SSL configuration',
             task: (ctx) => {
-                return argv.sslemail ? Promise.resolve({email: argv.sslemail}) : this.ui.prompt({
+                let promise = argv.sslemail ? Promise.resolve(argv.sslemail) : this.ui.prompt({
                     name: 'email',
                     type: 'input',
                     message: 'Enter your email (used for SSL certificate generation)',
                     validate: value => Boolean(value) || 'You must supply an email'
-                }).then((answer) => {
-                    argv.sslemail = answer.email;
+                }).then(answer => { argv.sslemail = answer.email; });
 
-                    ctx.ssl = {};
+                return Promise.all([
+                    promise,
+                    Promise.fromCallback((cb) => NginxConfFile.create(nginxConfPath, cb)).then((conf) => {
```

Comment:
I have a feeling that using Promise.all is going to lead to more problems?

I'm not 100% sure, but as promise.all doesn't guarantee any order, I guess this code could result in the ssl config having the well-known entry, but not having an email.

If there's an error from the prompt & the config got updated already... what happens if you try to run this again?